lib/config/fileconf_test.go

```diff
@@ -129,3 +129,38 @@ func (s *FileTestSuite) TestLegacyAuthenticationSection(c *check.C) {
 	c.Assert(fc.Auth.U2F.Facets, check.HasLen, 1)
 	c.Assert(fc.Auth.U2F.Facets[0], check.Equals, "https://graviton:3080")
 }
+
+// TestAppName makes sure application names are valid subdomains.
+func (s *FileTestSuite) TestAppName(c *check.C) {
+	tests := []struct {
+		desc     check.CommentInterface
+		inName   string
+		outValid bool
+	}{
+		{
+			desc:     check.Commentf("valid subdomain"),
+			inName:   "foo",
+			outValid: true,
+		},
+		{
+			desc:     check.Commentf("subdomains can not start with a dash"),
+			inName:   "-foo",
+			outValid: false,
+		},
+		{
+			desc:     check.Commentf("subdomains can not contain !"),
```

Comment:
`!` -> exclamation mark
just for clarity since the symbol is ambiguous in this context (e.g. the message could be referring to the whitespace and being extra passionate about it).